### PR TITLE
Opt app/improve multi mat thick

### PR DESCRIPTION
**Pull request by RezaNajian**
This PR improves and cleans implementations of multi-material and multi-thickness optimization. Some bug fixes, as well as test updates, are also done.

### CODE REVIEW

**Comment by RezaNajian** [comment]
@sunethwarna could you please review this PR ?!

File: applications/OptimizationApplication/custom_controls/thickness_controls/helmholtz_thickness.h [hunk 1]
```diff
@@ -558,17 +556,22 @@ class KRATOS_API(OPTIMIZATION_APPLICATION) HelmholtzThickness : public Thickness
...
+            pow_val = -700;        
+
+        // if(index_x1>0){
+        //     double prev_x1,prev_x2,prev_y1,prev_y2;
+        //     prev_x1 = x_limits[index_x1-1];
+        //     prev_x2 = x_limits[index_x1];
+        //     prev_y1 = y_limits[index_x1-1];
+        //     prev_y2 = y_limits[index_x1];
+        //     double prev_pow_val = -2.0*beta*(x1-(prev_x1+prev_x2)/2);
+        //     y1 = (prev_y2-prev_y1)/(1+std::exp(prev_pow_val)) + prev_y1;
```
Comment (sunethwarna) [inline applications/OptimizationApplication/custom_controls/thickness_controls/helmholtz_thickness.h]: I would remove the commented blocks

File: applications/OptimizationApplication/custom_python/optimization_python_application.cpp [hunk 2]
```diff
@@ -123,11 +123,13 @@ PYBIND11_MODULE(KratosOptimizationApplication, m)
 
     // thickness control
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, PT);
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, PPT);
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, CT);
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, FT);    
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, D_CT);  
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, D_PT); 
     KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, D_PT_D_FT); 
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, D_PPT_D_FT);
```
Comment (sunethwarna) [inline applications/OptimizationApplication/custom_python/optimization_python_application.cpp:132]: As we discussed, we need to write some explanations of these variables.

File: applications/OptimizationApplication/custom_responses/structural/stress_opt_response.h [hunk 3]
```diff
@@ -257,14 +257,14 @@ class KRATOS_API(OPTIMIZATION_APPLICATION) StressOptResponse : public Response
...
-            elem_i.GetProperties().SetValue(YOUNG_MODULUS,re_pr);
-        }
+        // if(elem_i.GetProperties().Has(E_PR)){
+        //     // double e_max = elem_i.GetProperties().GetValue(E_MAX);
+        //     // double e_min = elem_i.GetProperties().GetValue(E_MIN);
+        //     // double e_pr = elem_i.GetProperties().GetValue(E_PR);
+        //     // double e_pe = elem_i.GetProperties().GetValue(E_PE);
+        //     // double re_pr = e_min + std::pow((e_pr-e_min)/(e_max-e_min),mStressPenaltyFactor) * (e_max-e_min);
+        //     // elem_i.GetProperties().SetValue(YOUNG_MODULUS,re_pr);
+        // }
```
Comment (sunethwarna) [inline applications/OptimizationApplication/custom_responses/structural/stress_opt_response.h]: remove the block

File: applications/OptimizationApplication/custom_responses/structural/stress_opt_response.h [hunk 4]
```diff
@@ -293,10 +293,10 @@ class KRATOS_API(OPTIMIZATION_APPLICATION) StressOptResponse : public Response
...
         }
 
-        if(elem_i.GetProperties().Has(E_PE)){
-            double e_pe = elem_i.GetProperties().GetValue(E_PE);
-            elem_i.GetProperties().SetValue(YOUNG_MODULUS,e_pe);
-        }
+        // if(elem_i.GetProperties().Has(E_PE)){
+        //     double e_pe = elem_i.GetProperties().GetValue(E_PE);
+        //     elem_i.GetProperties().SetValue(YOUNG_MODULUS,e_pe);
+        // }
```
Comment (sunethwarna) [inline applications/OptimizationApplication/custom_responses/structural/stress_opt_response.h]: remove the block

**Review by sunethwarna — CHANGES_REQUESTED** [review]
Thanks for the patience @RezaNajian. I like the additions. I have mentioned few concerns in the PR files. Basically the followings:
1. Could you please remove thecommented blocks?
2. Could you please break all the header only classes to .h and .cpp files?

File: applications/OptimizationApplication/custom_python/optimization_python_application.cpp (hunk 2 above)
Comment (RezaNajian) [reply applications/OptimizationApplication/custom_python/optimization_python_application.cpp:132]: @sunethwarna I will make a separate PR for making h, cpp for classes and also for giving description of variables. Agreed ?!

**Review by sunethwarna — APPROVED** [review]
Approving, as agreed  followings needs to be addressed in future PRs.
1. Breaking all the header only files to .h. and .cpp
2. Giving descriptions for different shortened and abbreviated variables.
3. Removing trailing white spaces from all the files
4. Following the Kratos style guide.

**Pull request merged**